tdms/tests/system/data/input_generation/generate_test_input.py

```diff
@@ -27,9 +62,41 @@ def run_bscan(
     The bscan/ and matlab/ directories are assumed to already be in the
     includepath of the engine instance, so that the run_bscan and supporting
     MATLAB files can be called.
+
+    The obstacle radius is the circular face radius for cylinders (cyl), sphere radius for spheres (sph), and is ignored by freespace (fs) and point-source (sc) obstacles.
+
+    :param input_filename: The path to the input file that defines the variables iteratefdtd_matrix reads in
+    :param engine: The MatlabEngine instance to call run_bscan within. If not provided, a new session will be started and ended once the call is complete.
+    :param obstacle: The obstacle that is present in the simulation.
+    :param obstacle_radius: Radius of the spatial obstacle in microns.
+    :param illsetup: Flags whether run_bscan requires a call to iteratefdtd_matrix in illsetup mode as well as filesetup mode.
     """
-    # function [] = run_bscan(test_directory, input_filename)
-    engine.run_bscan(str(test_directory), str(input_filename), nargout=0)
+    # In the event that illsetup is required for this run, generate the temporary name for the input file to be passed to iteratefdtd_matrix in filesetup mode
+    # This only occurs when illumination files are required in input-data regeneration.
+    illfile_extra_file = ""
+    if illsetup:
+        # If this is _not_ an empty string, we need to generate the illumination file from the input file
+        # It is essentially identical, but needs the efname and hfname variables to be:
+        # present in the workspace/file when calling with 'illsetup'
+        # absent from the workspace/file when calling with 'filesetup'
+        # Otherwise, the "input" file to the illsetup and input file for the .mat creation are identical. As such, the optimal way to get around this is to have Python pass the input_file via 'illsetup', then copy this file, remove the definition of the efname & hfname variables, then run 'filesetp' mode. We can then cleanup our "extra" file that we created.
+        illfile_extra_file = (
+            os.path.splitext(input_filename)[0] + "temp_filesetup_file____.m"
+        )
+        _create_temporary_filesetup(input_filename, illfile_extra_file)
```

Comment:
I [still think](https://github.com/UCL/TDMS/pull/241#discussion_r1165227854) you can trim these doc comments, but I don't insist.

Reply:
I believe they will trim themselves further up the chain. (not a satisfying resolution I know)